### PR TITLE
Interactable Drawings

### PR DESCRIPTION
**Pull request by parseccentric**
# _REPLACE ME: Pull Request Description_

## _REPLACE ME: Name of feature / bug fix / etc._

_REPLACE ME: summary of the change(s)_

_REPLACE ME: motivation and context_

_REPLACE ME: Fixes #9999999_

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Testing 
- [ ] Change that requires a documentation update

### How Has This Been Tested?

_REPLACE ME: Action to take -> Expected result in own client. Expected result in other client. Example: Draw a stroke -> Own client should see stroke. Other client should see same stroke with same color, position, thickness._ 

_REPLACE ME: any relevant details for your test configuration_

- [ ] Manual Test
- [ ] Unit Test
- [ ] Integration / End-to-End Test

[Duplicate all of the above for each distinct feature / bug fix / etc.]
_____

# Test Configuration

* _REPLACE ME: Browser vendor and version: e.g., Chrome Version 91.0.4472.164 (Official Build) (64-bit)_
* _REPLACE ME: VR Device and OS version: e.g., Oculus Quest v29_
- [ ] PC VR (Link / AirLink)
- [ ] Standalone 

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas that are not self-documenting
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings 
- [ ] My changes have no unnecessary logging
- [ ] I have added tests that prove my fix is effective or that my feature works, for sufficiently complex features
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] Sensitive info like tokens, secrets, and passwords have been removed before submitting

Modified from this article:
Phillip Johnston, “A GitHub Pull Request Template for Your Projects - Embedded Artistry,” Embedded Artistry, Aug. 04, 2017. https://embeddedartistry.com/blog/2017/08/04/a-github-pull-request-template-for-your-projects/ (accessed Jul. 22, 2021).

### CODE REVIEW

**Comment by Barasakar** [comment]
Some errors happened during the testing
image

**Comment by parseccentric** [comment]
Tested:

* Draw individually and try to translate, rotate, and scale own strokes
* Draw individually and try to translate, rotate, and scale each other's strokes
* Erase everything and make sure nothing is left in either client
* Undo erasing and try to grab own strokes
* Undo erasing and try to grab each other's strokes

All manual tests passed.

File: Komodo/Assets/Packages/KomodoCore/Hidden~/WebGLTemplates/KomodoWebXRFullView2020/relay.js [hunk 1]
```diff
@@ -1,19 +1,31 @@
...
 var RELAY_BASE_URL = "http://localhost:3000";
 var API_BASE_URL = "http://localhost:4040";
-var VR_BASE_URL = "https://localhost:8123"; //TODO -- change this to a better default
+var VR_BASE_URL = "http://localhost:8123"; //TODO -- change this to a better default
 
 // init globals which Unity will assign when setup is done.
 var sync = null;
 var chat = null;
 
+/*
```
Comment (parseccentric) [inline Komodo/Assets/Packages/KomodoCore/Hidden~/WebGLTemplates/KomodoWebXRFullView2020/relay.js:37]: put this above the global variables.

File: Komodo/Assets/WebGLTemplates/KomodoWebXRFullView2020/relay.js [hunk 2]
```diff
@@ -15,6 +20,12 @@ var VR_BASE_URL = "http://localhost:8123"; //TODO -- change this to a better def
 var sync = null;
 var chat = null;
 
+/*
```
Comment (parseccentric) [inline Komodo/Assets/WebGLTemplates/KomodoWebXRFullView2020/relay.js:38]: TODO: move this above the global sync and chat variables

**Comment by parseccentric** [comment]
In Chrome with Admin user:

```komodo-unity-v0.5.8-2022-02-07-1446.framework.js:3616 Exception: Index is out-of-bounds for the client's networked game objects list.
  at Komodo.Runtime.NetworkedObjectsManager.GetNetworkedGameObject (System.Int32 buttonId) [0x00000] in <00000000000000000000000000000000>:0 


_JS_Log_Dump @ komodo-unity-v0.5.8-2022-02-07-1446.framework.js:3616
$__Z19WebGLPrintfConsolev7LogTypePKcPi @ 0e0ee9d2:0x28c69a
$__ZL20InternalErrorConsolePKcz @ 0e0ee9d2:0x28c516
$__Z40DebugStringToFilePostprocessedStacktraceRK21DebugStringToFileData @ 0e0ee9d2:0x28c0ac
$__Z17DebugStringToFileRK21DebugStringToFileData @ 0e0ee9d2:0x28a3c5
$__ZN9Scripting23LogExceptionFromManagedE21ScriptingExceptionPtriPKcb @ 0e0ee9d2:0x2a2aa6
$__ZN9Scripting12LogExceptionE21ScriptingExceptionPtriPKcb @ 0e0ee9d2:0x2d2193
$__ZN13MonoBehaviour30InvokeMethodOrCoroutineCheckedE18ScriptingMethodPtr18ScriptingObjectPtr @ 0e0ee9d2:0x4397d2
$__ZN9Scripting20SendScriptingMessageER10GameObjectPKc18ScriptingObjectPtr @ 0e0ee9d2:0x60a198
$_SendMessageString @ 0e0ee9d2:0x51de0f
(anonymous) @ komodo-unity-v0.5.8-2022-02-07-1446.framework.js:25664
ccall @ komodo-unity-v0.5.8-2022-02-07-1446.framework.js:507
SendMessage @ komodo-unity-v0.5.8-2022-02-07-1446.framework.js:156
SendMessage @ komodo-unity-v0.5.8-2022-02-07-1446.loader.js:95
(anonymous) @ komodo-unity-v0.5.8-2022-02-07-1446.framework.js:4332
r.emit @ index.js:83
r.onevent @ index.js:83
r.onpacket @ index.js:83
(anonymous) @ index.js:83
r.emit @ index.js:83
r.ondecoded @ index.js:83
(anonymous) @ index.js:83
r.emit @ index.js:83
a.add @ index.js:83
r.ondata @ index.js:83
(anonymous) @ index.js:83
r.emit @ index.js:83
r.onPacket @ index.js:83
(anonymous) @ index.js:83
r.emit @ index.js:83
r.onPacket @ index.js:83
r.onData @ index.js:83
ws.onmessage @ index.js:83
Show 3 more frames
komodo-unity-v0.5.8-2022-02-07-1446.loader.js:441          POST https://cdp.cloud.unity3d.com/v1/events net::ERR_BLOCKED_BY_CLIENT
CachedXMLHttpRequest.send @ komodo-unity-v0.5.8-2022-02-07-1446.loader.js:441
_JS_WebRequest_Send @ komodo-unity-v0.5.8-2022-02-07-1446.framework.js:3944
$__ZN19TransportJavaScript9DoRequestERK27TransportDoRequestArgsProtoI12HeaderHelper15DownloadHandler13UploadHandler18CertificateHandler14ResponseHelperE @ 0e0ee9d2:0xd70f25
$__ZN20UnityWebRequestProtoI24UnityWebRequestTransport16AtomicRefCounter14RedirectHelper14ResponseHelper15DownloadHandler13UploadHandler18CertificateHandler12HeaderHelper14AsyncOperationE9DoRequestEv @ 0e0ee9d2:0xd6f9af
$__ZN20UnityWebRequestProtoI24UnityWebRequestTransport16AtomicRefCounter14RedirectHelper14ResponseHelper15DownloadHandler13UploadHandler18CertificateHandler12HeaderHelper14AsyncOperationE14Task_DoRequestEPv @ 0e0ee9d2:0xd6e9e2
$__ZN20UnityWebRequestProtoI24UnityWebRequestTransport16AtomicRefCounter14RedirectHelper14ResponseHelper15DownloadHandler13UploadHandler18CertificateHandler12HeaderHelper14AsyncOperationE26Job_ExecuteUnityWebRequestEPS9_ @ 0e0ee9d2:0xd6eb84
$__ZN18BackgroundJobQueue19ScheduleJobInternalEPFvPvES0_RKNS_8JobFenceEN8JobQueue16JobQueuePriorityE @ 0e0ee9d2:0x458c0f
$__ZN20UnityWebRequestProtoI24UnityWebRequestTransport16AtomicRefCounter14RedirectHelper14ResponseHelper15DownloadHandler13UploadHandler18CertificateHandler12HeaderHelper14AsyncOperationE5BeginEv @ 0e0ee9d2:0xd6e993
$__ZN15UnityWebRequest5BeginEv @ 0e0ee9d2:0xd6e6ce
$__ZN11UnityEngine9Analytics20WebRequestRestClient11PerformPostEv @ 0e0ee9d2:0xd30eb3
$__ZN11UnityEngine9Analytics14DataDispatcher18PerformWebDispatchERNS1_9DataBlockERNS0_20WebRequestRestClientE @ 0e0ee9d2:0xd377a4
$__ZN11UnityEngine9Analytics14DataDispatcher23PerformRetryWebDispatchEv @ 0e0ee9d2:0xd3712a
$__ZN11UnityEngine9Analytics14DataDispatcher30HandleDispatcherProcessingStepEv @ 0e0ee9d2:0xd36a1b
$__ZN17DispatcherService17PlayerLoopHandlerEv @ 0e0ee9d2:0xd342a4
$__ZZN18AnalyticsCoreStats23RegisterGlobalCallbacksEvEN46EarlyUpdateAnalyticsCoreStatsUpdateRegistrator7ForwardEv @ 0e0ee9d2:0xd3418f
$__Z17ExecutePlayerLoopP22NativePlayerLoopSystem @ 0e0ee9d2:0x521643
$__Z17ExecutePlayerLoopP22NativePlayerLoopSystem @ 0e0ee9d2:0x521658
$__Z10PlayerLoopv @ 0e0ee9d2:0x520666
$__ZL8MainLoopv @ 0e0ee9d2:0x51e6d0
$__Z28MainLoopUpdateFromBackgroundPv @ 0e0ee9d2:0x546cbf
$dynCall_vi @ 0e0ee9d2:0x2716f63
(anonymous) @ komodo-unity-v0.5.8-2022-02-07-1446.framework.js:29589
dynCall @ komodo-unity-v0.5.8-2022-02-07-1446.framework.js:440
dynCall_wrapper @ komodo-unity-v0.5.8-2022-02-07-1446.framework.js:423
wrapper @ komodo-unity-v0.5.8-2022-02-07-1446.framework.js:3579
(anonymous) @ komodo-unity-v0.5.8-2022-02-07-1446.framework.js:10372
5komodo-unity-v0.5.8-2022-02-07-1446.framework.js:3616 Exception: Index is out-of-bounds for the client's networked game objects list.
  at Komodo.Runtime.NetworkedObjectsManager.GetNetworkedGameObject (System.Int32 buttonId) [0x00000] in <00000000000000000000000000000000>:0 


_JS_Log_Dump @ komodo-unity-v0.5.8-2022-02-07-1446.framework.js:3616
$__Z19WebGLPrintfConsolev7LogTypePKcPi @ 0e0ee9d2:0x28c69a
$__ZL20InternalErrorConsolePKcz @ 0e0ee9d2:0x28c516
$__Z40DebugStringToFilePostprocessedStacktraceRK21DebugStringToFileData @ 0e0ee9d2:0x28c0ac
$__Z17DebugStringToFileRK21DebugStringToFileData @ 0e0ee9d2:0x28a3c5
$__ZN9Scripting23LogExceptionFromManagedE21ScriptingExceptionPtriPKcb @ 0e0ee9d2:0x2a2aa6
$__ZN9Scripting12LogExceptionE21ScriptingExceptionPtriPKcb @ 0e0ee9d2:0x2d2193
$__ZN13MonoBehaviour30InvokeMethodOrCoroutineCheckedE18ScriptingMethodPtr18ScriptingObjectPtr @ 0e0ee9d2:0x4397d2
$__ZN9Scripting20SendScriptingMessageER10GameObjectPKc18ScriptingObjectPtr @ 0e0ee9d2:0x60a198
$_SendMessageString @ 0e0ee9d2:0x51de0f
(anonymous) @ komodo-unity-v0.5.8-2022-02-07-1446.framework.js:25664
ccall @ komodo-unity-v0.5.8-2022-02-07-1446.framework.js:507
SendMessage @ komodo-unity-v0.5.8-2022-02-07-1446.framework.js:156
SendMessage @ komodo-unity-v0.5.8-2022-02-07-1446.loader.js:95
(anonymous) @ komodo-unity-v0.5.8-2022-02-07-1446.framework.js:4332
r.emit @ index.js:83
r.onevent @ index.js:83
r.onpacket @ index.js:83
(anonymous) @ index.js:83
r.emit @ index.js:83
r.ondecoded @ index.js:83
(anonymous) @ index.js:83
r.emit @ index.js:83
a.add @ index.js:83
r.ondata @ index.js:83
(anonymous) @ index.js:83
r.emit @ index.js:83
r.onPacket @ index.js:83
(anonymous) @ index.js:83
r.emit @ index.js:83
r.onPacket @ index.js:83
r.onData @ index.js:83
ws.onmessage @ index.js:83
Show 3 more frames
komodo-unity-v0.5.8-2022-02-07-1446.loader.js:441          POST https://cdp.cloud.unity3d.com/v1/events net::ERR_BLOCKED_BY_CLIENT
CachedXMLHttpRequest.send @ komodo-unity-v0.5.8-2022-02-07-1446.loader.js:441
_JS_WebRequest_Send @ komodo-unity-v0.5.8-2022-02-07-1446.framework.js:3944
$__ZN19TransportJavaScript9DoRequestERK27TransportDoRequestArgsProtoI12HeaderHelper15DownloadHandler13UploadHandler18CertificateHandler14ResponseHelperE @ 0e0ee9d2:0xd70f25
$__ZN20UnityWebRequestProtoI24UnityWebRequestTransport16AtomicRefCounter14RedirectHelper14ResponseHelper15DownloadHandler13UploadHandler18CertificateHandler12HeaderHelper14AsyncOperationE9DoRequestEv @ 0e0ee9d2:0xd6f9af
$__ZN20UnityWebRequestProtoI24UnityWebRequestTransport16AtomicRefCounter14RedirectHelper14ResponseHelper15DownloadHandler13UploadHandler18CertificateHandler12HeaderHelper14AsyncOperationE14Task_DoRequestEPv @ 0e0ee9d2:0xd6e9e2
$__ZN20UnityWebRequestProtoI24UnityWebRequestTransport16AtomicRefCounter14RedirectHelper14ResponseHelper15DownloadHandler13UploadHandler18CertificateHandler12HeaderHelper14AsyncOperationE26Job_ExecuteUnityWebRequestEPS9_ @ 0e0ee9d2:0xd6eb84
$__ZN18BackgroundJobQueue19ScheduleJobInternalEPFvPvES0_RKNS_8JobFenceEN8JobQueue16JobQueuePriorityE @ 0e0ee9d2:0x458c0f
$__ZN20UnityWebRequestProtoI24UnityWebRequestTransport16AtomicRefCounter14RedirectHelper14ResponseHelper15DownloadHandler13UploadHandler18CertificateHandler12HeaderHelper14AsyncOperationE5BeginEv @ 0e0ee9d2:0xd6e993
$__ZN15UnityWebRequest5BeginEv @ 0e0ee9d2:0xd6e6ce
$__ZN11UnityEngine9Analytics20WebRequestRestClient11PerformPostEv @ 0e0ee9d2:0xd30eb3
$__ZN11UnityEngine9Analytics14DataDispatcher18PerformWebDispatchERNS1_9DataBlockERNS0_20WebRequestRestClientE @ 0e0ee9d2:0xd377a4
$__ZN11UnityEngine9Analytics14DataDispatcher23PerformRetryWebDispatchEv @ 0e0ee9d2:0xd3712a
$__ZN11UnityEngine9Analytics14DataDispatcher30HandleDispatcherProcessingStepEv @ 0e0ee9d2:0xd36a1b
$__ZN17DispatcherService17PlayerLoopHandlerEv @ 0e0ee9d2:0xd342a4
$__ZZN18AnalyticsCoreStats23RegisterGlobalCallbacksEvEN46EarlyUpdateAnalyticsCoreStatsUpdateRegistrator7ForwardEv @ 0e0ee9d2:0xd3418f
$__Z17ExecutePlayerLoopP22NativePlayerLoopSystem @ 0e0ee9d2:0x521643
$__Z17ExecutePlayerLoopP22NativePlayerLoopSystem @ 0e0ee9d2:0x521658
$__Z10PlayerLoopv @ 0e0ee9d2:0x520666
$__ZL8MainLoopv @ 0e0ee9d2:0x51e6d0
$__Z28MainLoopUpdateFromBackgroundPv @ 0e0ee9d2:0x546cbf
$dynCall_vi @ 0e0ee9d2:0x2716f63
(anonymous) @ komodo-unity-v0.5.8-2022-02-07-1446.framework.js:29589
dynCall @ komodo-unity-v0.5.8-2022-02-07-1446.framework.js:440
dynCall_wrapper @ komodo-unity-v0.5.8-2022-02-07-1446.framework.js:423
wrapper @ komodo-unity-v0.5.8-2022-02-07-1446.framework.js:3579
(anonymous) @ komodo-unity-v0.5.8-2022-02-07-1446.framework.js:10372
5komodo-unity-v0.5.8-2022-02-07-1446.framework.js:3616 Exception: Index is out-of-bounds for the client's networked game objects list.
  at Komodo.Runtime.NetworkedObjectsManager.GetNetworkedGameObject (System.Int32 buttonId) [0x00000] in <00000000000000000000000000000000>:0 


_JS_Log_Dump @ komodo-unity-v0.5.8-2022-02-07-1446.framework.js:3616
$__Z19WebGLPrintfConsolev7LogTypePKcPi @ 0e0ee9d2:0x28c69a
$__ZL20InternalErrorConsolePKcz @ 0e0ee9d2:0x28c516
$__Z40DebugStringToFilePostprocessedStacktraceRK21DebugStringToFileData @ 0e0ee9d2:0x28c0ac
$__Z17DebugStringToFileRK21DebugStringToFileData @ 0e0ee9d2:0x28a3c5
$__ZN9Scripting23LogExceptionFromManagedE21ScriptingExceptionPtriPKcb @ 0e0ee9d2:0x2a2aa6
$__ZN9Scripting12LogExceptionE21ScriptingExceptionPtriPKcb @ 0e0ee9d2:0x2d2193
$__ZN13MonoBehaviour30InvokeMethodOrCoroutineCheckedE18ScriptingMethodPtr18ScriptingObjectPtr @ 0e0ee9d2:0x4397d2
$__ZN9Scripting20SendScriptingMessageER10GameObjectPKc18ScriptingObjectPtr @ 0e0ee9d2:0x60a198
$_SendMessageString @ 0e0ee9d2:0x51de0f
(anonymous) @ komodo-unity-v0.5.8-2022-02-07-1446.framework.js:25664
ccall @ komodo-unity-v0.5.8-2022-02-07-1446.framework.js:507
SendMessage @ komodo-unity-v0.5.8-2022-02-07-1446.framework.js:156
SendMessage @ komodo-unity-v0.5.8-2022-02-07-1446.loader.js:95
(anonymous) @ komodo-unity-v0.5.8-2022-02-07-1446.framework.js:4332
r.emit @ index.js:83
r.onevent @ index.js:83
r.onpacket @ index.js:83
(anonymous) @ index.js:83
r.emit @ index.js:83
r.ondecoded @ index.js:83
(anonymous) @ index.js:83
r.emit @ index.js:83
a.add @ index.js:83
r.ondata @ index.js:83
(anonymous) @ index.js:83
r.emit @ index.js:83
r.onPacket @ index.js:83
(anonymous) @ index.js:83
r.emit @ index.js:83
r.onPacket @ index.js:83
r.onData @ index.js:83
ws.onmessage @ index.js:83
Show 3 more frames
komodo-unity-v0.5.8-2022-02-07-1446.loader.js:441          POST https://cdp.cloud.unity3d.com/v1/events net::ERR_BLOCKED_BY_CLIENT
CachedXMLHttpRequest.send @ komodo-unity-v0.5.8-2022-02-07-1446.loader.js:441
_JS_WebRequest_Send @ komodo-unity-v0.5.8-2022-02-07-1446.framework.js:3944
$__ZN19TransportJavaScript9DoRequestERK27TransportDoRequestArgsProtoI12HeaderHelper15DownloadHandler13UploadHandler18CertificateHandler14ResponseHelperE @ 0e0ee9d2:0xd70f25
$__ZN20UnityWebRequestProtoI24UnityWebRequestTransport16AtomicRefCounter14RedirectHelper14ResponseHelper15DownloadHandler13UploadHandler18CertificateHandler12HeaderHelper14AsyncOperationE9DoRequestEv @ 0e0ee9d2:0xd6f9af
$__ZN20UnityWebRequestProtoI24UnityWebRequestTransport16AtomicRefCounter14RedirectHelper14ResponseHelper15DownloadHandler13UploadHandler18CertificateHandler12HeaderHelper14AsyncOperationE14Task_DoRequestEPv @ 0e0ee9d2:0xd6e9e2
$__ZN20UnityWebRequestProtoI24UnityWebRequestTransport16AtomicRefCounter14RedirectHelper14ResponseHelper15DownloadHandler13UploadHandler18CertificateHandler12HeaderHelper14AsyncOperationE26Job_ExecuteUnityWebRequestEPS9_ @ 0e0ee9d2:0xd6eb84
$__ZN18BackgroundJobQueue19ScheduleJobInternalEPFvPvES0_RKNS_8JobFenceEN8JobQueue16JobQueuePriorityE @ 0e0ee9d2:0x458c0f
$__ZN20UnityWebRequestProtoI24UnityWebRequestTransport16AtomicRefCounter14RedirectHelper14ResponseHelper15DownloadHandler13UploadHandler18CertificateHandler12HeaderHelper14AsyncOperationE5BeginEv @ 0e0ee9d2:0xd6e993
$__ZN15UnityWebRequest5BeginEv @ 0e0ee9d2:0xd6e6ce
$__ZN11UnityEngine9Analytics20WebRequestRestClient11PerformPostEv @ 0e0ee9d2:0xd30eb3
$__ZN11UnityEngine9Analytics14DataDispatcher18PerformWebDispatchERNS1_9DataBlockERNS0_20WebRequestRestClientE @ 0e0ee9d2:0xd377a4
$__ZN11UnityEngine9Analytics14DataDispatcher23PerformRetryWebDispatchEv @ 0e0ee9d2:0xd3712a
$__ZN11UnityEngine9Analytics14DataDispatcher30HandleDispatcherProcessingStepEv @ 0e0ee9d2:0xd36a1b
$__ZN17DispatcherService17PlayerLoopHandlerEv @ 0e0ee9d2:0xd342a4
$__ZZN18AnalyticsCoreStats23RegisterGlobalCallbacksEvEN46EarlyUpdateAnalyticsCoreStatsUpdateRegistrator7ForwardEv @ 0e0ee9d2:0xd3418f
$__Z17ExecutePlayerLoopP22NativePlayerLoopSystem @ 0e0ee9d2:0x521643
$__Z17ExecutePlayerLoopP22NativePlayerLoopSystem @ 0e0ee9d2:0x521658
$__Z10PlayerLoopv @ 0e0ee9d2:0x520666
$__ZL8MainLoopv @ 0e0ee9d2:0x51e6d0
$__Z28MainLoopUpdateFromBackgroundPv @ 0e0ee9d2:0x546cbf
$dynCall_vi @ 0e0ee9d2:0x2716f63
(anonymous) @ komodo-unity-v0.5.8-2022-02-07-1446.framework.js:29589
dynCall @ komodo-unity-v0.5.8-2022-02-07-1446.framework.js:440
dynCall_wrapper @ komodo-unity-v0.5.8-2022-02-07-1446.framework.js:423
wrapper @ komodo-unity-v0.5.8-2022-02-07-1446.framework.js:3579
(anonymous) @ komodo-unity-v0.5.8-2022-02-07-1446.framework.js:10372
3komodo-unity-v0.5.8-2022-02-07-1446.framework.js:3616 Exception: Index is out-of-bounds for the client's networked game objects list.
  at Komodo.Runtime.NetworkedObjectsManager.GetNetworkedGameObject (System.Int32 buttonId) [0x00000] in <00000000000000000000000000000000>:0 
```

**Comment by parseccentric** [comment]
- [x] TODO: relay.js for loop -- put curly braces on same line
- [x] TODO: relay.js remove extraneous console logs
- [x] TODO: DIM whitespace, comments, regions, variables cleanup

**Comment by parseccentric** [comment]
@Barasakar Thanks! I will make a release branch for this and try to get as far as I can in addressing the above bugs.

**Comment by Barasakar** [comment]
@parseccentric No problem! Sorry about the late merge.

**Comment by parseccentric** [comment]
It's ok! I was running behind myself.

**Pull request merged**